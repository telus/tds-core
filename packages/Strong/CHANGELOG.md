# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## 2.2.3 (2022-02-16)


### Bug Fixes

* gha release job npm publish token ([ce7aacd](https://github.com/telusdigital/tds/commit/ce7aacdd77e54c4f8e48eb31d0c05dc804530324))





## 2.2.2 (2022-02-15)


### Bug Fixes

* gha release job git config ([276ebb6](https://github.com/telusdigital/tds/commit/276ebb6968a0b56c9b87c178f6895a60ae108e71))





## 2.2.1 (2022-02-15)


### Bug Fixes

* gha release job git config ([1ab603d](https://github.com/telusdigital/tds/commit/1ab603d68c36219b0711fc353bc2515b64712ca9))





# [2.2.0](https://github.com/telusdigital/tds/compare/@tds/core-strong@2.1.14...@tds/core-strong@2.2.0) (2021-09-21)


### Features

* **core-strong:** add react 17 to peer deps ([03f9b9b](https://github.com/telusdigital/tds/commit/03f9b9beb3336eb6dffbc7747b3df571397754d3))





## [2.1.14](https://github.com/telusdigital/tds/compare/@tds/core-strong@2.1.13...@tds/core-strong@2.1.14) (2021-01-15)

**Note:** Version bump only for package @tds/core-strong





## [2.1.13](https://github.com/telusdigital/tds/compare/@tds/core-strong@2.1.12...@tds/core-strong@2.1.13) (2020-09-11)

**Note:** Version bump only for package @tds/core-strong





## [2.1.12](https://github.com/telusdigital/tds/compare/@tds/core-strong@2.1.11...@tds/core-strong@2.1.12) (2020-09-08)


### Bug Fixes

* **core-strong:** add support for styled-components v5 ([c9664cc](https://github.com/telusdigital/tds/commit/c9664cc69839495d80411ecf01fbe98524ddc12e))





## [2.1.11](https://github.com/telusdigital/tds/compare/@tds/core-strong@2.1.10...@tds/core-strong@2.1.11) (2020-09-04)

**Note:** Version bump only for package @tds/core-strong





## [2.1.10](https://github.com/telusdigital/tds/compare/@tds/core-strong@2.1.9...@tds/core-strong@2.1.10) (2020-02-03)

**Note:** Version bump only for package @tds/core-strong





## [2.1.9](https://github.com/telusdigital/tds/compare/@tds/core-strong@2.1.8...@tds/core-strong@2.1.9) (2020-01-27)

**Note:** Version bump only for package @tds/core-strong





## [2.1.8](https://github.com/telusdigital/tds/compare/@tds/core-strong@2.1.7...@tds/core-strong@2.1.8) (2020-01-23)

**Note:** Version bump only for package @tds/core-strong





## [2.1.7](https://github.com/telusdigital/tds/compare/@tds/core-strong@2.1.6...@tds/core-strong@2.1.7) (2020-01-22)

**Note:** Version bump only for package @tds/core-strong





## [2.1.6](https://github.com/telusdigital/tds/compare/@tds/core-strong@2.1.5...@tds/core-strong@2.1.6) (2019-10-09)

**Note:** Version bump only for package @tds/core-strong





## [2.1.5](https://github.com/telusdigital/tds/compare/@tds/core-strong@2.1.4...@tds/core-strong@2.1.5) (2019-10-04)

**Note:** Version bump only for package @tds/core-strong





## [2.1.4](https://github.com/telusdigital/tds/compare/@tds/core-strong@2.1.3...@tds/core-strong@2.1.4) (2019-09-06)

**Note:** Version bump only for package @tds/core-strong





## [2.1.3](https://github.com/telusdigital/tds/compare/@tds/core-strong@2.1.2...@tds/core-strong@2.1.3) (2019-08-22)

**Note:** Version bump only for package @tds/core-strong





## [2.1.2](https://github.com/telusdigital/tds/compare/@tds/core-strong@2.1.1...@tds/core-strong@2.1.2) (2019-08-09)

**Note:** Version bump only for package @tds/core-strong





## [2.1.1](https://github.com/telusdigital/tds/compare/@tds/core-strong@2.1.0...@tds/core-strong@2.1.1) (2019-08-01)

**Note:** Version bump only for package @tds/core-strong





# [2.1.0](https://github.com/telusdigital/tds/compare/@tds/core-strong@2.0.2...@tds/core-strong@2.1.0) (2019-07-02)


### Features

* **core-strong:** add sup tag support ([315f78c](https://github.com/telusdigital/tds/commit/315f78c))





## [2.0.2](https://github.com/telusdigital/tds/compare/@tds/core-strong@2.0.1...@tds/core-strong@2.0.2) (2019-06-04)

**Note:** Version bump only for package @tds/core-strong

## [2.0.1](https://github.com/telusdigital/tds/compare/@tds/core-strong@2.0.0...@tds/core-strong@2.0.1) (2019-05-01)

**Note:** Version bump only for package @tds/core-strong

# [2.0.0](https://github.com/telusdigital/tds/compare/@tds/core-strong@1.0.9...@tds/core-strong@2.0.0) (2019-03-28)

### Features

- **core-strong:** upgrade to styled-components ([1ddec6b](https://github.com/telusdigital/tds/commit/1ddec6b))

### BREAKING CHANGES

- **core-strong:** upgrade to styled-components

## [1.0.9](https://github.com/telusdigital/tds/compare/@tds/core-strong@1.0.8...@tds/core-strong@1.0.9) (2019-03-20)

**Note:** Version bump only for package @tds/core-strong

## [1.0.8](https://github.com/telusdigital/tds/compare/@tds/core-strong@1.0.7...@tds/core-strong@1.0.8) (2019-03-20)

**Note:** Version bump only for package @tds/core-strong

## [1.0.7](https://github.com/telusdigital/tds/compare/@tds/core-strong@1.0.6...@tds/core-strong@1.0.7) (2019-03-18)

**Note:** Version bump only for package @tds/core-strong

## [1.0.6](https://github.com/telusdigital/tds/compare/@tds/core-strong@1.0.5...@tds/core-strong@1.0.6) (2019-03-05)

**Note:** Version bump only for package @tds/core-strong

## [1.0.5](https://github.com/telusdigital/tds/compare/@tds/core-strong@1.0.4...@tds/core-strong@1.0.5) (2019-02-06)

**Note:** Version bump only for package @tds/core-strong

<a name="1.0.4"></a>

## [1.0.4](https://github.com/telusdigital/tds/compare/@tds/core-strong@1.0.3...@tds/core-strong@1.0.4) (2018-12-19)

**Note:** Version bump only for package @tds/core-strong

<a name="1.0.3"></a>

## [1.0.3](https://github.com/telusdigital/tds/compare/@tds/core-strong@1.0.2...@tds/core-strong@1.0.3) (2018-12-06)

**Note:** Version bump only for package @tds/core-strong

<a name="1.0.2"></a>

## [1.0.2](https://github.com/telusdigital/tds/compare/@tds/core-strong@1.0.1...@tds/core-strong@1.0.2) (2018-07-10)

**Note:** Version bump only for package @tds/core-strong

<a name="1.0.1"></a>

## [1.0.1](https://github.com/telusdigital/tds/compare/@tds/core-strong@1.0.0...@tds/core-strong@1.0.1) (2018-06-14)

**Note:** Version bump only for package @tds/core-strong
