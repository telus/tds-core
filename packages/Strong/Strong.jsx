import React from 'react'
import PropTypes from 'prop-types'

import safeRest from '../../shared/utils/safeRest'

import styles from '../Text/Text.modules.scss'

/**
 * Give portions of a sentence added importance.
 */
const Strong = ({ children, ...rest }) => (
  <strong {...safeRest(rest)} className={styles.boldFont}>
    {children}
  </strong>
)

Strong.propTypes = {
  /**
   * The text.
   */
  children: PropTypes.node.isRequired,
}

export default Strong
