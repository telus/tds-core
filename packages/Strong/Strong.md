```
<Text>
  <Strong>Your password is not valid.</Strong> It must not be one of your previously used passwords.
</Text>
```
