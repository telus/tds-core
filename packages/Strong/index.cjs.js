const Strong = require('./dist/index.cjs')

module.exports = Strong
