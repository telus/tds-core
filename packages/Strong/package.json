{
  "name": "@tds/core-strong",
  "version": "0.0.0-development",
  "description": "Give portions of a sentence added importance",
  "main": "index.cjs.js",
  "module": "index.es.js",
  "license": "MIT",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/telusdigital/tds.git"
  },
  "publishConfig": {
    "access": "public"
  },
  "author": "TELUS digital",
  "engines": {
    "node": ">=8"
  },
  "bugs": {
    "url": "https://github.com/telusdigital/tds/issues"
  },
  "homepage": "http://tds.telus.com",
  "peerDependencies": {
    "react": ">=15",
    "react-dom": ">=15"
  },
  "dependencies": {
    "prop-types": "^15.5.10"
  },
  "devDependencies": {
    "@tds/core-colours": "^0.0.0-development"
  }
}
