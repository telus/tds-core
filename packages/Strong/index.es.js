import Strong from './dist/index.es'

export default Strong
