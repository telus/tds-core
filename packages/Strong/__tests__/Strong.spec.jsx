import React from 'react'
import { shallow } from 'enzyme'

import Strong from '../Strong'

describe('Strong', () => {
  const doShallow = props => shallow(<Strong {...props}>Some content</Strong>)

  it('renders', () => {
    const strong = doShallow()

    expect(strong).toMatchSnapshot()
  })

  it('renders an HTML strong tag', () => {
    const strong = doShallow()

    expect(strong).toHaveDisplayName('strong')
  })

  it('passes additional attributes to the strong element', () => {
    const strong = doShallow({ id: 'the-id' })

    expect(strong).toHaveProp('id', 'the-id')
  })

  it('does not allow custom CSS', () => {
    const strong = doShallow({ className: 'my-custom-class', style: { color: 'hotpink' } })

    expect(strong).not.toHaveProp('className', 'my-custom-class')
    expect(strong).not.toHaveProp('style')
  })
})
