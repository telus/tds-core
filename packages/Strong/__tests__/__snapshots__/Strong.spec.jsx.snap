// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Strong renders 1`] = `
<strong
  className="boldFont"
>
  Some content
</strong>
`;
